tests/conftest.py

```diff
@@ -69,31 +87,34 @@ def mock_gridded_obs_dataset():
         },
         coords={"time": time, "latitude": latitudes, "longitude": longitudes},
     )
+    # Set a specific value for a specific time and location to remove ambiguity
+    dataset["2m_temperature"].loc[
+        dict(
+            time="2021-06-21 18:00",
+            latitude=slice(40, 45),
+            longitude=slice(100, 105),
+        )
+    ] = 25
     return dataset
 
 
 @pytest.fixture
-def mock_gridded_obs_dataset_max_in_forecast():
-    time = pd.date_range("2021-06-20", freq="3h", periods=200)
-    data = np.random.RandomState(21897820).standard_normal(size=(len(time), 180, 360))
-    data[10, :, :] = 5
-    latitudes = np.linspace(-90, 90, 180)
-    longitudes = np.linspace(0, 359, 360)
-    dataset = xr.Dataset(
-        {
-            "2m_temperature": (["time", "latitude", "longitude"], 20 + 5 * data),
-            "tp": (["time", "latitude", "longitude"], data),
-            "10m_u_component_of_wind": (["time", "latitude", "longitude"], data),
-            "10m_v_component_of_wind": (["time", "latitude", "longitude"], data),
-        },
-        coords={"time": time, "latitude": latitudes, "longitude": longitudes},
+def mock_forecast_dataarray(mock_forecast_dataset):
+    return dataset_to_dataarray(mock_forecast_dataset)
```

Comment:
I'm not sure if it's kosher to compose fixtures like this... I generally avoid it.

I would recommend defining the actual `xr.Dataset` in a stand-alone method, "make_sample_forecast_dataset() -> xr.Dataset". Then, your fixture `sample_forecast_dataset` can just call this function and return it naked, and this "sample_forecast_dataarray" fixture can call it and return "dataset_to_datarray" applied to it - no need to pass in the dataset fixture, which will make life in the actual unit test implementation more managable.